mypyc/irbuild/util.py

```diff
@@ -41,19 +48,37 @@ def is_trait(cdef: ClassDef) -> bool:
 
 def dataclass_decorator_type(d: Expression) -> str | None:
     if isinstance(d, RefExpr) and d.fullname in DATACLASS_DECORATORS:
-        return d.fullname.split(".")[0]
+        name = d.fullname.split(".")[0]
+        if name == "attrs":
+            raise ValueError("Slotted attrs classes are not supported")
```

Comment:
Ideally this would use the builder error plumbing, but TBF mypyc's error reporting story is pretty terrible anyway (see "usability" under https://github.com/mypyc/mypyc/issues/785).

Reply:
It's not really acceptable to raise `ValueError`. This should report an error using the `mypyc.errors.Errors` class.